### ISSUE TITLE
[core] Regression: JavaClass not found by ClassNode

### ISSUE THREAD

**Issue by ljrk0** [skylot/jadx#1276]
**Checks before report**
- [x] check [Troubleshooting Q&A](https://github.com/skylot/jadx/wiki/Troubleshooting-Q&A) section on wiki
- [x] search existing issues by exception message

**Describe error**
Error when decompiling `com.redacted.foo.R$string`

With the following stack trace:
```
$ gradle :jadx-cli:run --args='-d redacted redacted.apk' 
INFO  - loading ...
INFO  - processing ...
ERROR - jadx error: JavaClass not found by ClassNode: com.redacted.foo.R$string
jadx.core.utils.exceptions.JadxRuntimeException: JavaClass not found by ClassNode: com.redacted.foo.R$string
        at jadx.api.JadxDecompiler.getJavaClassByNode(JadxDecompiler.java:462)
        at jadx.core.utils.Utils.collectionMapNoNull(Utils.java:216)
        at jadx.core.utils.DecompilerScheduler.lambda$buildBatches$0(DecompilerScheduler.java:37)
        at jadx.core.utils.Utils.collectionMap(Utils.java:205)
        at jadx.core.utils.DecompilerScheduler.buildBatches(DecompilerScheduler.java:37)
        at jadx.api.JadxDecompiler.appendSourcesSave(JadxDecompiler.java:299)
        at jadx.api.JadxDecompiler.getSaveTasks(JadxDecompiler.java:269)
        at jadx.api.JadxDecompiler.getSaveExecutor(JadxDecompiler.java:227)
        at jadx.api.JadxDecompiler.getSaveExecutor(JadxDecompiler.java:215)
        at jadx.api.JadxDecompiler.save(JadxDecompiler.java:180)
        at jadx.cli.JadxCLI.processAndSave(JadxCLI.java:48)
        at jadx.cli.JadxCLI.execute(JadxCLI.java:35)
        at jadx.cli.JadxCLI.main(JadxCLI.java:19)
```

Unfortunately I cannot provide the APK nor the full class names due to an NDA.  However I can provide the following additional information and am willing to provide a patch myself, I may require some assistance around the Jadx code base though.

The regression was introduced in commit 570e7528a7eb084ee202551f42fafe31e7d37f25, between 1.2.0 and master:

```
$ git bisect log
git bisect start
# bad: [985ccd6bba2ec4f38d063ed918e6ff8b59f71a74] feat: save jobf when decompiling to Java (#1274) (PR #1275)
git bisect bad 985ccd6bba2ec4f38d063ed918e6ff8b59f71a74
# good: [381afa27410249aae4a99a3f88bb65ecce548f6c] chore: fix release config
git bisect good 381afa27410249aae4a99a3f88bb65ecce548f6c
# skip: [c52c659b9464ccbbb1cf23f01f92a3d59097130e] fix: correct inline flag for variables used in anonymous classes (#1154)
git bisect skip c52c659b9464ccbbb1cf23f01f92a3d59097130e
# skip: [1462acbb9211057cb5727b7dc1d3a63660b0c5e2] chore: remove not needed file
git bisect skip 1462acbb9211057cb5727b7dc1d3a63660b0c5e2
# skip: [6bf358fc6694c7cd3596c28af0541542a63780db] feat(gui): improve exclude package feature (#1151) (PR #1152)
git bisect skip 6bf358fc6694c7cd3596c28af0541542a63780db
# good: [5f24193c49bbbd460231e9818923d6fa1cf0ec7b] chore: update dependencies
git bisect good 5f24193c49bbbd460231e9818923d6fa1cf0ec7b
# good: [3a298122411ccbe21ee87940ae8ef9c90e7e986d] fix(gui): resolve NPE during index (#1254)
git bisect good 3a298122411ccbe21ee87940ae8ef9c90e7e986d
# good: [dfdc14ea8618a7242d04bd6ea066d3e6b60a10b9] feat: rename without deobfuscation, save renames in project (#1076 #1022)
git bisect good dfdc14ea8618a7242d04bd6ea066d3e6b60a10b9
# good: [17f99ed92861bdf863ad9666ed3614794236fc76] fix: adjust class processing order for correct methods inline (#1264)
git bisect good 17f99ed92861bdf863ad9666ed3614794236fc76
# good: [5fc27c11362b37d56d5f7fc466d53d14da82cbf0] perf(gui): improve decompilation speed (#1269)
git bisect good 5fc27c11362b37d56d5f7fc466d53d14da82cbf0
# good: [918585968d5f613636c3b2253c465a7cbc8adca0] perf(gui): on rename unload dependent classes instead recompile
git bisect good 918585968d5f613636c3b2253c465a7cbc8adca0

< now at 570e7528a7eb084ee202551f42fafe31e7d37f25 which fails >
```

However, the commit 5fc27c11362b37d56d5f7fc466d53d14da82cbf0 already introduced a similar issue which went away with 918585968d5f613636c3b2253c465a7cbc8adca0 again.  The similar issue produced the following trace:

```
INFO  - loading ...
INFO  - processing ...
INFO  - progress: 2095 of 9310 (22%)
ERROR - Error saving class: com.redacted.foo.C0219R$xml
jadx.core.utils.exceptions.JadxRuntimeException: Failed to generate code for class: com.redacted.foo.R
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:116)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:300)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:265)
        at jadx.api.JavaClass.getCodeInfo(JavaClass.java:54)
        at jadx.api.JadxDecompiler.lambda$appendSourcesSave$4(JadxDecompiler.java:303)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: jadx.core.utils.exceptions.JadxRuntimeException: Code generation error
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:55)
        at jadx.core.codegen.CodeGen.generateJavaCode(CodeGen.java:34)
        at jadx.core.codegen.CodeGen.generate(CodeGen.java:22)
        at jadx.core.ProcessClass.process(ProcessClass.java:68)
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:110)
        ... 7 common frames omitted
Caused by: java.util.ConcurrentModificationException: null
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1631)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at jadx.core.dex.attributes.nodes.JadxCommentsAttr.formatAndFilter(JadxCommentsAttr.java:36)
        at jadx.core.utils.CodeGenUtils.addComments(CodeGenUtils.java:53)
        at jadx.core.utils.CodeGenUtils.addErrorsAndComments(CodeGenUtils.java:28)
        at jadx.core.codegen.ClassGen.addClassCode(ClassGen.java:128)
        at jadx.core.codegen.ClassGen.makeClass(ClassGen.java:94)
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:45)
        ... 11 common frames omitted
INFO  - progress: 2129 of 9310 (22%)
ERROR - Error saving class: com.redacted.foo.C0205R$color
jadx.core.utils.exceptions.JadxRuntimeException: Failed to generate code for class: com.redacted.foo.R
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:116)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:300)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:265)
        at jadx.api.JavaClass.getCodeInfo(JavaClass.java:54)
        at jadx.api.JadxDecompiler.lambda$appendSourcesSave$4(JadxDecompiler.java:303)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: jadx.core.utils.exceptions.JadxRuntimeException: Code generation error
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:55)
        at jadx.core.codegen.CodeGen.generateJavaCode(CodeGen.java:34)
        at jadx.core.codegen.CodeGen.generate(CodeGen.java:22)
        at jadx.core.ProcessClass.process(ProcessClass.java:68)
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:110)
        ... 7 common frames omitted
Caused by: java.util.ConcurrentModificationException: null
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1631)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at jadx.core.dex.attributes.nodes.JadxCommentsAttr.formatAndFilter(JadxCommentsAttr.java:36)
        at jadx.core.utils.CodeGenUtils.addComments(CodeGenUtils.java:53)
        at jadx.core.utils.CodeGenUtils.addErrorsAndComments(CodeGenUtils.java:28)
        at jadx.core.codegen.ClassGen.addClassCode(ClassGen.java:128)
        at jadx.core.codegen.ClassGen.makeClass(ClassGen.java:94)
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:45)
        ... 11 common frames omitted
ERROR - Error saving class: com.redacted.foo.C0215R$plurals
jadx.core.utils.exceptions.JadxRuntimeException: Failed to generate code for class: com.redacted.foo.R
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:116)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:300)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:265)
        at jadx.api.JavaClass.getCodeInfo(JavaClass.java:54)
        at jadx.api.JadxDecompiler.lambda$appendSourcesSave$4(JadxDecompiler.java:303)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: jadx.core.utils.exceptions.JadxRuntimeException: Code generation error
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:55)
        at jadx.core.codegen.CodeGen.generateJavaCode(CodeGen.java:34)
        at jadx.core.codegen.CodeGen.generate(CodeGen.java:22)
        at jadx.core.ProcessClass.process(ProcessClass.java:68)
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:110)
        ... 7 common frames omitted
Caused by: java.util.ConcurrentModificationException: null
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1631)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at jadx.core.dex.attributes.nodes.JadxCommentsAttr.formatAndFilter(JadxCommentsAttr.java:36)
        at jadx.core.utils.CodeGenUtils.addComments(CodeGenUtils.java:53)
        at jadx.core.utils.CodeGenUtils.addErrorsAndComments(CodeGenUtils.java:28)
        at jadx.core.codegen.ClassGen.addClassCode(ClassGen.java:128)
        at jadx.core.codegen.ClassGen.makeClass(ClassGen.java:94)
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:45)
        ... 11 common frames omitted
INFO  - progress: 9287 of 9310 (99%)
ERROR - finished with errors, count: 44
```

Note that jadx is complaining about not being able to generate code for the same class that with a later commit it will complain about not finding;  however only `$xml` etc. and not `$string`.

Maybe it is also noteworthy that with the 'original' regression jadx still finished and printed the progress, while with newer commit it aborts completely.  Before the regressions where introduced, I still had decompilation errors but no exceptions logged to the console, e.g. with 17f99ed92861bdf863ad9666ed3614794236fc76:

```
INFO  - loading ...
INFO  - processing ...
INFO  - progress: 10718 of 10773 (99%)
ERROR - finished with errors, count: 34
```

Anyway, from what I can tell the commit in question add another check https://github.com/skylot/jadx/blob/master/jadx-core/src/main/java/jadx/api/JadxDecompiler.java#L438 for `javaClass.getClassNode() == cls` which evaluates to false in this case, thus throwing an exception in line 462.  Since `Utils::collectionMapNoNull` doesn't catch exceptions it will simply fail as soon as such an exception occurs.
This means however, that the regressing commit maybe only highlighted the problem and not introduced it.

So far, I would at first simply go with adding some exception handling here to continue decompilation.
But I'm not sure where to handle the exception, either have `map` catch exceptions, collect them, and re-raise them or never call map with an exception?

**Comment by ljrk0** [comment]
I just ran `git bisect` starting at 5f24193c49bbbd460231e9818923d6fa1cf0ec7b as the last good commit and applying only the below diff on each bisection step:

```
diff --git a/jadx-core/src/main/java/jadx/api/JadxDecompiler.java b/jadx-core/src/main/java/jadx/api/JadxDecompiler.java
--- a/jadx-core/src/main/java/jadx/api/JadxDecompiler.java
+++ b/jadx-core/src/main/java/jadx/api/JadxDecompiler.java
@@ -419,7 +419,7 @@ public final class JadxDecompiler implements Closeable {
 	@ApiStatus.Internal
 	public JavaClass getJavaClassByNode(ClassNode cls) {
 		JavaClass javaClass = classesMap.get(cls);
-		if (javaClass != null) {
+		if (javaClass != null && javaClass.getClassNode() == cls) {
 			return javaClass;
 		}
 		// load parent class if inner
```

And this confirms that indeed 5fc27c11362b37d56d5f7fc466d53d14da82cbf0 is the commit that introduced the issue next to the `Error saving class` exceptions.  The latter were fixed later-on, while the former was hidden until 570e7528a7eb084ee202551f42fafe31e7d37f25 made it surface.

**Comment by ljrk0** [comment]
It seems that `buildBatches()` which tries to turn a `List<JavaClass>` into a `List<List<JavaClass>>` is faulty as it suddenly calls `decompiler::getJavaClassByNode` on `ClassNode`s whose `JavaClass`' `ClassNode` isn't the same as the original `ClassNode`, i.e., where

    f(cls) {
        return classesMap.get(cls).getClassNode()
    }

is not the identity function.  From the annotation above `getJavaClassByNode` I could imagine this being generated classes which should be skipped in the call to `collectionMapNoNull`?  I've tried pursuing this as a workaround basing my code on `ExportGradleProject::skipGeneratedClasses`:

```diff
diff --git a/jadx-core/src/main/java/jadx/core/utils/DecompilerScheduler.java b/jadx-core/src/main/java/jadx/core/utils/DecompilerScheduler.java
index 146c17c4..380b0e33 100644
--- a/jadx-core/src/main/java/jadx/core/utils/DecompilerScheduler.java
+++ b/jadx-core/src/main/java/jadx/core/utils/DecompilerScheduler.java
@@ -1,7 +1,9 @@
 package jadx.core.utils;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
+import java.util.stream.Collectors;
 import java.util.Comparator;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -30,11 +32,14 @@ public class DecompilerScheduler implements IDecompileScheduler {
 		this.decompiler = decompiler;
 	}
 
+	private static final Set<String> IGNORE_CLS_NAMES = new HashSet<>(Arrays.asList("R", "BuildConfig"));
+
 	@Override
 	public List<List<JavaClass>> buildBatches(List<JavaClass> classes) {
 		long start = System.currentTimeMillis();
 		List<List<ClassNode>> batches = internalBatches(Utils.collectionMap(classes, JavaClass::getClassNode));
-		List<List<JavaClass>> result = Utils.collectionMap(batches, l -> Utils.collectionMapNoNull(l, decompiler::getJavaClassByNode));
+		List<List<ClassNode>> ngBatches = Utils.collectionMap(batches, b -> b.stream().filter(cls -> IGNORE_CLS_NAMES.contains(cls.getClassInfo().getShortName())).collect(Collectors.toList()));
+		List<List<JavaClass>> result = Utils.collectionMap(ngBatches, l -> Utils.collectionMapNoNull(l, decompiler::getJavaClassByNode));
 		if (LOG.isDebugEnabled()) {
 			LOG.debug("Build decompilation batches in {}ms", System.currentTimeMillis() - start);
 		}
```

This indeed fixes the above bug:

```
ERROR - jadx error: JavaClass not found by ClassNode: com.redacted.foo.R$string
```

but introduces:

```
INFO  - loading ...
INFO  - processing ...
INFO  - progress: 2085 of 9310 (22%)
ERROR - Error saving class: com.redacted.foo.R
jadx.core.utils.exceptions.JadxRuntimeException: Failed to generate code for class: com.redacted.foo.R
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:120)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:319)
        at jadx.core.dex.nodes.ClassNode.decompile(ClassNode.java:265)
        at jadx.api.JavaClass.getCodeInfo(JavaClass.java:54)
        at jadx.api.JadxDecompiler.lambda$appendSourcesSave$4(JadxDecompiler.java:303)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: jadx.core.utils.exceptions.JadxRuntimeException: Code generation error
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:55)
        at jadx.core.codegen.CodeGen.generateJavaCode(CodeGen.java:34)
        at jadx.core.codegen.CodeGen.generate(CodeGen.java:22)
        at jadx.core.ProcessClass.process(ProcessClass.java:72)
        at jadx.core.ProcessClass.generateCode(ProcessClass.java:114)
        ... 7 common frames omitted
Caused by: java.util.ConcurrentModificationException: null
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1631)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
        at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
        at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
        at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
        at jadx.core.dex.attributes.nodes.JadxCommentsAttr.formatAndFilter(JadxCommentsAttr.java:36)
        at jadx.core.utils.CodeGenUtils.addComments(CodeGenUtils.java:53)
        at jadx.core.utils.CodeGenUtils.addErrorsAndComments(CodeGenUtils.java:28)
        at jadx.core.codegen.ClassGen.addClassCode(ClassGen.java:128)
        at jadx.core.codegen.ClassGen.makeClass(ClassGen.java:94)
        at jadx.core.codegen.CodeGen.wrapCodeGen(CodeGen.java:45)
        ... 11 common frames omitted
INFO  - progress: 9309 of 9310 (99%)
INFO  - done
```

So, if we simply filter those nodes we do not get a working decompilation, likely, because these classes are needed to decompile the "main" classes.  So maybe, in fact the actual bug is introduced in 570e7528a7eb084ee202551f42fafe31e7d37f25 as this gui-bug-fix has unintended side-effects?  But I'm just second-guessing here.

**Comment by skylot** [comment]
@ljrk0 fixed in 4ee4a343236ecd0ec12dfb1b4a1c30ae6df47160.
Cause was in duplicated ClassNode object for same class (`R$raw`). That additional class was created by jadx to restore `R` resource class with inner types, but it was unexpected for me that `R` class can be missing but inner `R.raw` class exists, and I didn't check for that.
Thank you for report and your work 👍🏽

**Comment by ljrk0** [comment]
Thank you very much for the quick response, this indeed fixed all issues for me!
